v0/destinations/sendgrid/util.js

```diff
@@ -0,0 +1,152 @@
+      keys.to.forEach(keyto => {
+        if (keyto.email) {
+          personalizationsArr.push(keyto);
+        }
+      });
+    } else {
+      logger.error(`item at index ${index} dropped. to field is mandatory`);
+    }
+    updatedPayload.personalizations[index].to = personalizationsArr;
+    if (keys.subject) {
+      updatedPayload.personalizations[index].subject = keys.subject;
+    }
+  });
+  if (payload.attachments) {
+    if (!payload.attachments.content || !payload.attachments.filename) {
+      updatedPayload.attachments = null;
+      logger.error("content and filename are required for attachments");
+    }
+    if (!isValidBase64(payload.attachments.content)) {
+      updatedPayload.attachments = null;
+      logger.error("content should be base64 encoded");
+    }
+  }
+  if (!payload.from.email) {
+    throw new CustomError("email is required inside from object", 400);
+  }
+  if (payload.categories) {
+    payload.categories.forEach((category, index) => {
+      if (typeof category !== "string") {
+        updatedPayload.categories[index] = String(category);
+      }
+    });
+    payload.categories.splice(10);
+  }
+  return updatedPayload;
+};
+
+const eventValidity = (Config, event) => {
+  let flag = false;
+  Config.eventNamesSettings.forEach(eventName => {
+    if (eventName.event && eventName.event.trim().length !== 0) {
+      if (eventName.event.trim().toLowerCase() === event) {
+        flag = true;
+      }
+    }
+  });
+  if (!flag) {
+    throw new CustomError(
+      "Event not configured on your Rudderstack dashboard",
+      400
+    );
+  }
+};
+
+const createList = Config => {
+  const asmList = [];
+  if (Config.groupsToDisplay && Config.groupsToDisplay.length > 0) {
+    Config.groupsToDisplay.forEach(groups => {
+      if (
+        groups.groupId.trim() &&
+        isNaN(Number(groups.groupId)) &&
+        Number.isInteger(Number(groups.groupId))
+      ) {
+        asmList.push(Number(groups.groupId));
+      }
+    });
+  }
+  return asmList;
+};
+
+const createContent = Config => {
+  const contentList = [];
+  if (Config.contents && Config.contents.length > 0) {
+    Config.contents.forEach((content, index) => {
+      if (content.type && content.value) {
+        contentList.push(content);
+      } else {
+        logger.error(
+          `item at index ${index} dropped. type and value are required fields`
+        );
+      }
+    });
+  }
+  return contentList;
+};
+
+const createAttachments = Config => {
+  const attachmentList = [];
+  if (Config.attachments && Config.attachments.length > 0) {
+    Config.attachments.forEach((attachment, index) => {
+      if (attachment.content && attachment.type) {
+        attachmentList.push(attachment);
+      } else {
+        logger.error(
+          `item at index ${index} dropped. content and type are required fields`
+        );
+      }
+    });
+  }
+  return attachmentList;
+};
+
+const constructFields = (iObj, payload) => {
```

Comment:
we should refactor this function. Instead of using multiple if blocks, can we put the list of expected keys as a string[], iterate over the array and populated the updatedPayload?